### PR TITLE
DAOS-11886 object: skip invalid object during migrate unpack (#10645)

### PR DESCRIPTION
**Pull request by wangdi1**
Skip invalid object during migrate unpack. Add some debugging information.

Signed-off-by: Di Wang <di.wang@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by liuxuezhao — APPROVED** [review]

File: src/object/srv_obj_migrate.c [hunk 1]
```diff
@@ -2212,6 +2212,11 @@ migrate_enum_unpack_cb(struct dss_enum_unpack_io *io, void *data)
 	int			rc = 0;
 	int			i;
 
+	if (!daos_oclass_is_valid(daos_obj_id2class(io->ui_oid.id_pub))) {
+		D_WARN("Skip invalid "DF_UOID".\n", DP_UOID(io->ui_oid));
```
Comment (liuxuezhao) [inline src/object/srv_obj_migrate.c:2216]: is there a reasonable possibility to get invalid oid?

Comment (wangdi1) [reply src/object/srv_obj_migrate.c:2216]: Not sure yet, but I am adding some assertion and debug message in this patch.

**Review by jolivier23 — COMMENTED** [review]
Need to add request_for_2.2 label to the ticket.   But the comment on the ticket confused me a bit because you said you haven't seen it there and then posted this patch.

**Comment by wangdi1** [comment]
>Need to add request_for_2.2 label to the ticket. But the comment on the ticket confused me a bit because you said you haven't seen it there and then posted this patch.

Yes, it did not see, but in theory, it should land to 2.2 anyway to avoid corruption.

**Comment by wangdi1** [comment]
looks like all tests had been run, not sure why not all green here.

**Comment by johannlombardi** [comment]
Ready for landing?

**Pull request merged**